### PR TITLE
Alerts controller porting

### PR DESCRIPTION
**Pull request by cesar-tonnoir**
To check with @ouranos => there is something weird in the specs with :impac_kpi_id

### CODE REVIEW

File: api/lib/mno_enterprise/concerns/controllers/jpi/v1/impac/alerts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  end
+
+  # GET /jpi/v1/impac/alerts
+  def index
+    # Tenant must have kpi_enabled set to true to discover alerts.
+    if current_tenant && current_tenant.kpi_enabled?
+      @alerts = current_user.alerts
+    else
+      # Remove all kpi alerts
+      current_user.alerts.each(&:destroy)
```
Comment (xaun) [inline api/lib/mno_enterprise/concerns/controllers/jpi/v1/impac/alerts_controller.rb]: We don't want to actually destroy the alerts here do we? What if I become KPI enabled, then you've nuked all my kpi's alerts? We just want to return alerts that are not related to KPIs I think. Allowing for future alerts that are not KPI related.

File: api/lib/mno_enterprise/concerns/controllers/jpi/v1/impac/alerts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  # 'included do' causes the included code to be evaluated in the
+  # context where it is included rather than being executed in the module's context
+  included do
+    respond_to :json
+  end
+
+  # GET /jpi/v1/impac/alerts
+  def index
+    # Tenant must have kpi_enabled set to true to discover alerts.
+    if current_tenant && current_tenant.kpi_enabled?
```
Comment (xaun) [inline api/lib/mno_enterprise/concerns/controllers/jpi/v1/impac/alerts_controller.rb]: We need to move the code into mnohub I think (check with Oli), as this is something that we need to control - as it's an open source library.

File: api/lib/mno_enterprise/concerns/controllers/jpi/v1/impac/alerts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      @alerts = []
+    end
+  end
+
+  # POST /jpi/v1/impac/kpis/:kpi_id/alerts
+  def create
+    return render_bad_request('attach alert to kpi', 'no alert specified') unless params.require(:alert)
+    return render_not_found('kpi') unless kpi_alert.kpi
+
+    authorize! :create, kpi_alert
```
Comment (xaun) [inline api/lib/mno_enterprise/concerns/controllers/jpi/v1/impac/alerts_controller.rb]: Can't you just do `authorize! :manage_impac, kpi_alert.kpi.dashboard` ?

Comment (cesar-tonnoir) [inline api/lib/mno_enterprise/concerns/controllers/jpi/v1/impac/alerts_controller.rb]: we want to start have more fine-tuned abilities: someone could be able to create alerts for himself, but not to create/delete widgets and dashboards. Plus, :manage_impac doesn't seem to be activated in the other controllers.

File: api/lib/mno_enterprise/concerns/controllers/jpi/v1/impac/alerts_controller.rb (hunk 1 above)
Comment (cesar-tonnoir) [inline api/lib/mno_enterprise/concerns/controllers/jpi/v1/impac/alerts_controller.rb]: ok for that

File: api/lib/mno_enterprise/concerns/controllers/jpi/v1/impac/alerts_controller.rb (hunk 3 above)
Comment (xaun) [inline api/lib/mno_enterprise/concerns/controllers/jpi/v1/impac/alerts_controller.rb]: Ok fair enough

**Pull request merged**